src/vs/code/electron-main/launch.ts

```diff
@@ -178,7 +179,7 @@ export class LaunchService implements ILaunchService {
 		}
 
 		// Start without file/folder arguments
-		else if (args._.length === 0 && (args['folder-uri'] || []).length === 0) {
+		else if (!hasArgs(args._) && !hasArgs(args['folder-uri']) && !hasArgs(args['file-uri'])) {
```

Comment:
@aeschli I find this hard to read/understand and think that support for URIs should just be a boolean CLI argument to enable it but should not tamper with the `args._` value.  For example:

`code remote://foo/bar`: in this case `args._` should have one entry `remote://foo/bar` but we treat that argument as file path
`code remote://foo/bar --enable-uris`: in this case `args._` should have one entry `remote://foo/bar` but we treat it as URI.

We should not change the parsing/reading of what we open from the command line by introducing a new CLI flag that handles this. Rather we should preserve the semantics of `args._` which is to indicate what is being passed over to Code for opening.

Reply:
It would be great to just have a boolean attribute that signals that all arguments are passed as URIs. But how will we be able to distinguish which arguments are file or folders?